File: packages/cogo-chat-sdk-ts/src/ide-hints.ts
export type IdeHints = {
  toast?: string;
  toast_ko?: string;
  toast_en?: string;
  toast_ru?: string;
  toast_th?: string;
  toast_ja?: string;
  open_file?: string;
  highlight?: { path: string; range: [number, number] };
  next_action?: string;
  [k: string]: unknown;
};

export function selectToast(hints: IdeHints | null | undefined, preferred: string = 'en'): string | undefined {
  if (!hints) return undefined;
  const lang = (preferred || 'en').toLowerCase();
  const map: Record<string, string | undefined> = {
    en: hints.toast_en ?? hints.toast,
    ko: hints.toast_ko ?? hints.toast,
    ru: hints.toast_ru ?? hints.toast,
    th: hints.toast_th ?? hints.toast,
    ja: hints.toast_ja ?? hints.toast,
  };
  return map[lang] ?? hints.toast;
}

File: packages/cogo-chat-sdk-flutter/lib/models/ide_hints.dart
class IdeHintToast {
  final String message;
  final String? lang;

  IdeHintToast({required this.message, this.lang});

  factory IdeHintToast.fromJson(Map<String, dynamic> json) => IdeHintToast(
        message: json['message']?.toString() ?? '',
        lang: json['lang']?.toString(),
      );
}

class IdeHints {
  final String? toast;
  final String? toastKo;
  final String? toastEn;
  final String? toastRu;
  final String? toastTh;
  final String? toastJa;
  final Map<String, dynamic>? extra;

  IdeHints({this.toast, this.toastKo, this.toastEn, this.toastRu, this.toastTh, this.toastJa, this.extra});

  factory IdeHints.fromJson(Map<String, dynamic> json) => IdeHints(
        toast: json['toast']?.toString(),
        toastKo: json['toast_ko']?.toString(),
        toastEn: json['toast_en']?.toString(),
        toastRu: json['toast_ru']?.toString(),
        toastTh: json['toast_th']?.toString(),
        toastJa: json['toast_ja']?.toString(),
        extra: json,
      );

  String? selectToast(String preferred) {
    final lang = (preferred.isNotEmpty ? preferred : 'en').toLowerCase();
    final map = <String, String?>{
      'en': toastEn ?? toast,
      'ko': toastKo ?? toast,
      'ru': toastRu ?? toast,
      'th': toastTh ?? toast,
      'ja': toastJa ?? toast,
    };
    return map[lang] ?? toast;
  }
}
